src/roadmap/v1/lifecycle/app_streams.py

```diff
@@ -287,21 +301,21 @@ async def systems_by_app_stream(
         if not installed_packages:
             missing["installed_packages"] += 1
 
-        # Store package name, os_major, and system ID for later processing outside the loop.
+        # Store package name, os_major, system ID and display name for later processing outside the loop.
         # This substantially reduces the time it takes for this function to return.
-        system_id = system["id"]
+        system_key = (system["id"], system["display_name"])
```

Comment:
This should be a `SystemInfo` object and the variable name should change to match. This isn't a key field.

Reply:
Thanks! Creating the `SystemInfo` right in this place is a great idea!